dm/pkg/binlog/event/util.go

```diff
@@ -408,6 +408,27 @@ func GetCharsetCodecByStatusVars(statusVars []byte) (encoding.Encoding, error) {
 	}
 }
 
+// GetTimezoneByStatusVars returns the timezone of upstream for "datetime" type alter table commands
+// For "ALTER TABLE ADD x datetime", not for "ALTER TABLE ADD x timestamp".
+func GetTimezoneByStatusVars(statusVars []byte, upstreamTZStr string) (string, error) {
+	vars, err := statusVarsToKV(statusVars)
+	b, ok := vars[QTimeZoneCode]
+
+	if !ok {
+		if err == nil {
+			// only happen when this is a dummy event generated by DM
+			err = fmt.Errorf("Timezone not found in status_vars %v", statusVars)
+		}
+		return "", err
+	}
+
+	s := string(b[1:])
```

Comment:
plz leave a comment about the structure of this value

Reply:
Done.